llvm/lib/SYCLLowerIR/SYCLProcessJointMatrix.cpp

```diff
@@ -0,0 +1,78 @@
+//===- SYCLProcessJointMatrix.cpp - SYCL Joint Matrix transformation Pass -===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// A transformation pass which mutates Joint Matrix builtin calls to make them
+// conformat with SPIR-V friendly LLVM IR specification.
+//
+//===----------------------------------------------------------------------===//
+
+#include "llvm/SYCLLowerIR/SYCLProcessJointMatrix.h"
+
+#include "llvm/IR/IRBuilder.h"
+
+using namespace llvm;
+
+namespace {
+
+static constexpr char ACCESS_CHAIN[] = "_Z19__spirv_AccessChain";
+static constexpr char MATRIX_TYPE[] = "spirv.CooperativeMatrixKHR";
+
+// This routine extracts spirv.CooperativeMatrixKHR target extension type
+// from sycl::joint_matrix class object if it's used in __spirv_AccessChain
+// function call. It's necessary because otherwise OpAccessChain indices would
+// be wrong.
+bool transformAccessChain(Function *F) {
```

Comment:
currently it's the only function, but I see few other candidates in the translator that might be moved into this pass (of cause if we decide to introduce it)